__tests__/app.test.js

```diff
@@ -212,3 +212,103 @@ describe("6. GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("7. POST /api/articles/:article_id/comments/", () => {
+  it("status:200, should responds with the new comment objects if user exists", () => {
+    const newPost = {
+      username: "butter_bridge",
+      body: "test body",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newPost)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            body: expect.any(String),
+            article_id: 1,
+            author: expect.any(String),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+
+  it("status:404, should responds with error message when user does not exists in database but still post comment", () => {
+    const newPost = {
+      username: "some_random_dude",
+      body: "test body",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newPost)
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("User Not Found");
+      });
+  });
+
+  it("status:400, should responds with error message when req body missing some data", () => {
+    const newPost = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newPost)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Not Null Violation");
```

Comment:
I would recommend your error message is something more generic like "bad request" or "malformed request body" - an error message like this begins to expose your data structure eg. using SQL and could invite malicious attempts to mess with it